src/winston3.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as TransportStream from 'winston-transport';
+
+import {LOGGING_TRACE_KEY as COMMON_TRACE_KEY, LoggingCommon} from './common';
+import * as types from './types/core';
+
+type Callback = (err: Error, apiResponse: {}) => void;
+
+export class LoggingWinston extends TransportStream {
+  private common: LoggingCommon;
+  constructor(options?: types.Options) {
+    options = options || {};
+
+    super({
+      level: options.level,
+    });
+
+    this.common = new LoggingCommon(options);
+  }
+
+  log({message, level, splat, stack, ...metadata}: LogArg, callback: Callback) {
+    if (stack) {
+      // this happens if someone calls.
+      // logger.error(new Error('boop'))
+      // winston 3 console logging produces this output which is astoundingly
+      // less useful
+      //   {"level":"error"}
+      message = message + ' ' + stack;
+    }
+
+    this.common.log(level, message, metadata || {}, callback);
+  }
+}
+
+export const LOGGING_TRACE_KEY = COMMON_TRACE_KEY;
```

Comment:
`winston2` has the `LOGGING_TRACE_KEY` as a static property on the class too. Was this a conscious decision to not do the same here?

Secondly, exporting the key here doesn't publicly export it. I think this needs to be exported from `index.ts`. Any thoughts on whether an export from the module is the better interface for this vs. static property on the class? Note that this is likely to be semver major if we differ from what we used to do before.

Reply:
so i copied the documentation
```
const winston = require('winston');
const {LoggingWinston} = require('@google-cloud/logging-winston');

// ...

winston.info('Log entry with custom trace value', {
  [LoggingWinston.LOGGING_TRACE_KEY]: 'custom-trace-value'
});
```
i can add it to all the other places as well

Reply:
exporting here does publicly export it
```
LoggingWinston.LOGGING_TRACE_KEY = common.LOGGING_TRACE_KEY;
exports.LoggingWinston = LoggingWinston;
exports.LOGGING_TRACE_KEY = common.LOGGING_TRACE_KEY;
```

Reply:
fixed the exports =)